### DISCUSSION TITLE
Calling unocss from Node

### DISCUSSION THREAD

**Discussion by npavkovic** [unocss/unocss#1438] [Q&A] [answered]
Greetings,

We're working with various jamstack site generators (such as Eleventy) that don't have a webpack build. We use  npm packages to process our HTML before outputting a final version. So, following the pattern we use for other transform utilities (autoprefixer, SASS, etc), it'd be great to write:

```
const unocss = require("unocss");

myHTML = <unprocessed html code for page>;
var styleBlock = unocss(myHTML);

```
...where styleBlock would (obviously) contain the unocss-generated CSS. While I can probably (?) come up with code to do this from the existing code for plugins, was wondering if this already exists somewhere in the repo.

**Comment by zyyv** [comment, marked as answer]
UnoCSS can work in anywhere that as a css engine, our unit tests is node env, you can check it.
You need use `createGenerator` to create unocss context to use.

e.g:  
```ts
import { createGenerator } from 'unocss'

const uno = createGenerator({...}) // set you unocss config

const output = await uno.generate(htmlCode)
```
